### PR TITLE
Add move PanelSurveys to a folder

### PR DESCRIPTION
**Pull request by ftarulla**
This PR is the branch-rebase (with conflicts resolved) continuation of PR #2031

Related #1981 

In this PR we code refactor 3 main components to add "Move PanelSurveys to folder":
- `SurveyIndex`
- `FolderShow`
- `PanelSurveyShow`

The code refactor was necessary to remove "flags" that made the components behaviour depend on the element it was showing (survey or panel survey)
On the other hand auxiliar components were added to make these components feel "React". For example in the 3 components we added:
- `<MainActions>`
- `<Title>`
- `<MainView>`
- `<FoldersGrid>`
- `<SurveysGrid>`
- `<PagingFooter>`

Resulting in a render structure looking like this:
```
return (
  <div>
    <MainActions ...>
      <Action ... />
      <Action ... />
    </MainActions>

    <MainView ...>
      <FoldersGrid .../>
      <SurveysGrid .../>
      <PagingFooter ... />
    </MainView>

    <ConfirmationModal ... />
    <ConfirmationModal ... />
  </div>
)
```

On the other hand we introduce another component: `PanelSurveyCard` along the already existing `SurveyCard`. The logic shared between both components is reified in the (also new) component `InnerSurveyCard`

This structure (and the functions related) are shared between the 3 main components `SurveyIndex`, `PanelSurveyShow` and `FolderShow`. A second "code refactor" pass is needed to remove the repeated logic

### Also in this PR
- Fix refresh issues in `SurveyIndex` and `FolderShow`

### CODE REVIEW

**Review by ysbaddaden — CHANGES_REQUESTED** [review]
That still looks great!

I'm not comfortable with introducing so many duplication in React, so maybe we should tackle #2038 as quickly as possible! Even before #2029 I think.

**I had some issues:**

I think they're all because the behavior changed in the new `main` branch: we no longer have global arrays with _all_ folders and _all_ panel surveys and _all_ surveys, but load them on demand for each page.

1. I can't create a new wave because of an access to `latestOccurrence` instead of `latestWave` in PanelSurveyShow.jsx (i.e. there are still some `Occurrence` instead of `Wave` in the React components); I suspect this is a problem from `main`;

2. I created a new PanelSurvey (at the root) and tried to move it into a Folder, I expected the PanelSurvey to disappear, but it was still visible. This is a mere refresh issue, the PanelSurvey was  properly moved to the destination Folder.

3. I have an error when moving the freshly created PanelSurvey (only one wave, not even ready to be started) from one Folder to another one or back at the root: `items[action.surveyId] is undefined` in `surveysActions.changeFolder()`.

   Again, this is a refresh issue; the PanelSurvey was moved to the other Folder.

File: web/static/js/components/surveys/PanelSurveyShow.jsx [hunk 1]
```diff
@@ -6,37 +6,49 @@ import * as surveysActions from '../../actions/surveys'
...
+  // componentWillMount() {
+  //   const { dispatch, projectId, panelSurvey, panelSurveyId } = this.props
 
-    if (!panelSurvey) {
-      dispatch(actions.fetchPanelSurvey(projectId, panelSurveyId))
-    }
+  //   if (!panelSurvey) {
+  //     dispatch(actions.fetchPanelSurvey(projectId, panelSurveyId))
+  //   }
+  // }
```
[the remark this answers is not in the document]
Comment (ftarulla) [reply web/static/js/components/surveys/PanelSurveyShow.jsx]: 🙈

**Review by ysbaddaden — APPROVED** [review]
Yay, the refresh issues have been fixed!

I just have one React warning while testing:

- Failed prop type: missing ] after element list in FolderCard (created by FoldersGrid) —I don't understand this one :confused: 

There are few other warnings, but they're not related to this PR.

**Comment by ftarulla** [comment]
> Yay, the refresh issues have been fixed!
> 
> I just have one React warning while testing:
> 
> * Failed prop type: missing ] after element list in FolderCard (created by FoldersGrid) —I don't understand this one 😕
> 
> There are few other warnings, but they're not related to this PR.

I got this warning (and I think it's the same problem)
```
Warning: Failed prop type: Unexpected identifier
    in FolderCard (created by FoldersGrid)
    in FoldersGrid (created by SurveyIndex)
...
```

It's due to the `PropTypes` definition in `FolderCard` for `onDelete` and `onRename`:
```
  static propTypes = {
    projectId: PropTypes.number,
    id: PropTypes.number,
    name: PropTypes.string,
    t: PropTypes.func,
    onDelete: Function,
    onRename: Function,
    readOnly: PropTypes.bool
  }
```

It should be:
```
  static propTypes = {
    projectId: PropTypes.number,
    id: PropTypes.number,
    name: PropTypes.string,
    t: PropTypes.func,
    onDelete: PropTypes.func,
    onRename: PropTypes.func,
    readOnly: PropTypes.bool
  }
```

I made the fix in the PR #2041 and I'm also fixing it here. Let me know if it solves the warning.

**Review by ysbaddaden — APPROVED** [review]

**Pull request merged**